test/expressions/new-expr/new-on-domain.good

```diff
@@ -0,0 +1,2 @@
+new-on-domain.chpl:1: In function 'main':
+new-on-domain.chpl:2: error: invalid use of 'new' on chpl__buildDomainRuntimeType
```

Comment:
In fairness to me, this is an existing message from a path I didn't change.  I added the
test to the 2 I had a hand in to get this path on the table.  I am open to ideas on how
to make it more elegant but I'd be tempted to do that in a follow up PR